src/Training.cpp

```diff
@@ -342,18 +341,73 @@ void Training::process_game(GameState& state, size_t& train_pos, int who_won,
         step.to_move = to_move;
         step.planes = get_planes(&state);
 
-        step.probabilities.resize(POTENTIAL_MOVES);
-        step.probabilities[move_idx] = 1.0f;
+        if (policies.size() == 0) {
+            step.probabilities[move_idx] = 1.0f;
+        } else {
+            step.probabilities = std::move(policies[counter]);
```

Comment:
This does not do what you think it does because it's a std::array, not a std::vector.

Reply:
Good catch. I thought that std::array would behave similarly to std::vector (takes O(1) time to move) and didn't think how std::move would work for an aggregate type. Let me change it back to std::vector.